File: cpp/3rd_Week/MariMencoba3/MainBangunan2.cpp
// #include "Arsitek.cpp"
// #include "PenjualBahanBangunan.cpp"
#include "Manusia.cpp"
#include "TokoBangunan.cpp"
#include "Semen.cpp"

using namespace std;

int main()
{
        Arsitek dataArsitek;
        TokoBangunan dataToko;
        PenjualBahanBangunan dataPenjual;
        Semen dataSemen;

        //memasukan data ke dalam object arsitek
        dataArsitek.setKodeArsitek("A1");
        dataArsitek.setNama("Rendy");
        dataArsitek.setKantor("Podomoro Group");
        dataArsitek.setTahunBekerja("2021");

        dataToko.setKodeToko("TK01");
        dataToko.setNamaToko("PD Sumber Makmur");
        dataToko.setAlamatToko("Jalan Setiabudi");

        dataPenjual.setNoKtp("3204280012");
        dataPenjual.setNama("Hj. Asep");
        dataPenjual.setNamaToko("PD Sumber Makmur");
        dataPenjual.setAlamatToko("Jalan Setiabudi");

        dataSemen.setKodeBahan("SMN01");
        dataSemen.setKategoriJenisBahan("Semen");
        dataSemen.setNamaSemen("Semen Tiga Roda");
        dataSemen.setKodePabrik("PB001");
        dataSemen.setKualitas("Sangat Baik");
        dataSemen.setHarga("40K/sak");

        //        print data
        cout << "Data" << endl;
        cout << "===============" << endl;
        cout << "Arsitek" << endl;
        cout << "---------------" << endl;
        cout << "Kode: " << dataArsitek.getKodeArsitek() << endl;
        cout << "Nama: "+dataArsitek.getNama() << endl;
        cout << "Kantor: "+dataArsitek.getKantor() << endl;
        cout << "Tahun bekerja: "+dataArsitek.getTahunBekerja() << endl;
        cout << "---------------" << endl;
        cout << "Toko Bangunan" << endl;
        cout << "---------------" << endl;
        cout << "Kode Toko: "+dataToko.getKodeToko() << endl;
        cout << "Nama Toko: "+dataToko.getNamaToko() << endl;
        cout << "Alamat Toko: "+dataToko.getAlamatToko() << endl;
        cout << "---------------" << endl;
        cout << "Penjual Bahan Bangunan" << endl;
        cout << "---------------" << endl;
        cout << "No KTP: "+dataPenjual.getNoKtp() << endl;
        cout << "Nama Penjual: "+dataPenjual.getNama() << endl;
        cout << "Nama Toko: "+dataPenjual.getNamaToko() << endl;
        cout << "Alamat Toko: "+dataPenjual.getAlamatToko() << endl;
        cout << "---------------" << endl;
        cout << "Bahan Bangunan" << endl;
        cout << "---------------" << endl;
        cout << "1. Semen" << endl;
        cout << "   - KodeBahan : "+dataSemen.getKodeBahan() << endl;
        cout << "   - Kategori Jenis Bahan : "+dataSemen.getKategoriJenisBahan() << endl;
        cout << "   - Nama : "+dataSemen.getNamaSemen() << endl;
        cout << "   - Kode Pabrik : "+dataSemen.getKodePabrik() << endl;
        cout << "   - Kualitas : "+dataSemen.getKualitas() << endl;
        cout << "   - Harga : "+dataSemen.getHarga() << endl;
        return 0;
}

File: java/javadpbo/src/com/sem4dpbo/week3/marimencoba3/MainBangunan2.java
package com.sem4dpbo.week3.marimencoba3;

import com.sem4dpbo.week3.marimencoba2.Genteng;

public class MainBangunan {
    public static void main(String[] args) {
        Arsitek dataArsitek = new Arsitek();     // intansiasi class arsitek

//        memasukan data ke dalam object arsitek
        dataArsitek.setKodeArsitek("A1");
        dataArsitek.setNamaArsitek("Rendy");
        dataArsitek.setKantor("Podomoro Group");
        dataArsitek.setTahunBekerja("2021");

//        TokoBangunan dataToko = new TokoBangunan();     //instansiasi object data toko menggunakan class Toko

        //memasukan data ke dalam objcet dataPenjual memakai inheritance

        PenjualBahanBangunan dataPenjual = new PenjualBahanBangunan();       //instansiasi class penjual bahan Bangunan
        dataPenjual.setKodeToko("TK01");
        dataPenjual.setNamaToko("PD Sumber Makmur");
        dataPenjual.setAlamatToko("Jalan Setiabudi");
        dataPenjual.setNamaPenjual("Hj. Asep");
        dataPenjual.setNoKtp("3204280012");

        Semen dataSemen = new Semen();
        dataSemen.setKodeBahan("SMN01");
        dataSemen.setKategoriJenisBahan("Semen");
        dataSemen.setNamaSemen("Semen Tiga Roda");
        dataSemen.setKodePabrik("PB001");
        dataSemen.setKualitas("Sangat Baik");
        dataSemen.setHarga("40K/sak");


//        print data
        System.out.println("Data");
        System.out.println("===============");
        System.out.println("Arsitek");
        System.out.println("---------------");
        System.out.println("Kode: "+dataArsitek.getKodeArsitek());
        System.out.println("Nama: "+dataArsitek.getNamaArsitek());
        System.out.println("Kantor: "+dataArsitek.getKantor());
        System.out.println("Tahun bekerja: "+dataArsitek.getTahunBekerja());
        System.out.println("---------------");
        System.out.println("Toko Bangunan");
        System.out.println("---------------");
        System.out.println("Kode Toko: "+dataPenjual.getKodeToko());
        System.out.println("Nama Toko: "+dataPenjual.getNamaToko());
        System.out.println("Alamat Toko: "+dataPenjual.getAlamatToko());
        System.out.println("---------------");
        System.out.println("Penjual Bahan Bangunan");
        System.out.println("---------------");
        System.out.println("Kode Toko: "+dataPenjual.getKodeToko());
        System.out.println("Nama Toko: "+dataPenjual.getNamaToko());
        System.out.println("Alamat Toko: "+dataPenjual.getAlamatToko());
        System.out.println("Nama Penjual: "+dataPenjual.getNamaPenjual());
        System.out.println("No KTP: "+dataPenjual.getNoKtp());
        System.out.println("---------------");
        System.out.println("Bahan Bangunan");
        System.out.println("---------------");
        System.out.println("1. Semen");
        System.out.println("   - KodeBahan : "+dataSemen.getKodeBahan());
        System.out.println("   - Kategori Jenis Bahan : "+dataSemen.getKategoriJenisBahan());
        System.out.println("   - Nama : "+dataSemen.getNamaSemen());
        System.out.println("   - Kode Pabrik : "+dataSemen.getKodePabrik());
        System.out.println("   - Kualitas : "+dataSemen.getKualitas());
        System.out.println("   - Harga : "+dataSemen.getHarga());
    }
}
